### PR TITLE
fix(routing): support serving static files from a route used by a route handler

### PR DESCRIPTION
**Pull request by Goldziher**
This PR fixes an issue with static routing where static file serving failed if a route handler was defined for the base path of the static file serving.

Additionally two fixes for tests are added.

### CODE REVIEW

File: tests/unit/test_openapi/test_parameters.py
Comment (provinzkraut) [inline tests/unit/test_openapi/test_parameters.py:1]: What lead to the change here?

**Review by provinzkraut — APPROVED** [review]
Nice! :heart:

File: tests/unit/test_openapi/test_parameters.py
Comment (Goldziher) [reply tests/unit/test_openapi/test_parameters.py:1]: they were failing locally. Reason is an update of polyfactory which included a change to the random seed.

Comment (provinzkraut) [reply tests/unit/test_openapi/test_parameters.py:1]: So the fix is to bump polyfactory? If yes, could you include that in this PR?

Comment (Goldziher) [reply tests/unit/test_openapi/test_parameters.py:1]: i did... dunno why its failing here, its working locally.

**Pull request merged**